### ISSUE TITLE
✨ [Feature] Add Has, WhereHas, orWhereHas, DoesntHave , WhereDoesntHave, orWhereDoesntHave for Orm

### ISSUE THREAD

**Issue by hwbrzzl** [goravel/goravel#373]
### Before feedback (在反馈之前)

- [X] There are no features that I want to submit in Issues (当前 Issues 中没有我将要提交的新功能)

### Describe Feature (描述功能)

Like: 
- https://laravel.com/docs/10.x/eloquent-relationships#querying-relationship-existence
- https://laravel.com/docs/10.x/eloquent-relationships#querying-relationship-absence

There is a related PR: https://github.com/goravel/framework/pull/406, but there are some problems with it, you can check the comments for more information.

**Comment by shayan-yousefi** [comment]
Hey there! I came across the closed PR and noticed it's a bit old—looks like a lot has changed in the code over the past 5 months. I'm still using it to get some ideas though. Thanks

**Comment by hwbrzzl** [comment]
Yes, the basic logic is good, but some problems need to be solved. You can make an investigation first.

**Comment by darakanoit** [comment]
is there any plans to add whereLike? https://laravel.com/docs/11.x/queries#additional-where-clauses

**Comment by hwbrzzl** [comment]
@darakanoit Yes, it's simple, welcome PR.

**Comment by zoryamba** [comment]
Is anybody working on this? I think, I could take it

**Comment by hwbrzzl** [comment]
> Is anybody working on this? I think, I could take it

Sure, thanks. Please check the existing PR first.

**Comment by zoryamba** [comment]
Reviewed the PR. Have concerns about adding the `errors` field and checking errors in each executing method. I believe it would be better to add condition on method call and handle errors on build step. The question here, should I add another field under Conditions struct, kinda
```Go
// database/gorm/conditions.go
type Conditions struct {
  ...
  where         []contractsdriver.Where
  whereHas      []contractsdriver.Where
  ...
}

```
and separate buildWhereHas method? 
Or should I add just another WhereType under `contracts/database/driver/conditions.go` and build it as a separate case in `buildWhere` method? How do you see it?

**Comment by hwbrzzl** [comment]
A new `whereHas` will break the sort of `where`, so it may not be a good choice.

**Comment by zoryamba** [comment]
So, I was able to reproduce working code from the PR. Also made some refactoring, so that condition is being build on a build phase. I'm still working on tests and other methods, so it's still in draft, but could you please review the code, am I moving the right way?

**Comment by hwbrzzl** [comment]
@zoryamba Sure, thanks, I'll check it.
